### DISCUSSION TITLE
关于证书（certificate）

### DISCUSSION THREAD

**Discussion by YuQiangZ** [kubeedge/kubeedge#4618] [Q&A] [answered]
在官网的文档中看到“--advertise-address（仅从1.3版本开始可用）是云端公开的地址（将添加到CloudCore证书的SAN中），默认值为本地IP。”
      我的理解是，--advertise-address指定的IP会出现在证书的SAN中，但是我在证书的SAN中并没有看到此IP。请问在哪儿可以查看到该IP已经放入SAN。

**Comment by vincentgoat** [comment]
The CloudCore cert is hosted by the secret, checking it through `kubectl get secret -nkubeedge cloudcoresecret -oyaml`. Transfer the field `data.cloudcoredata` to cert format and then decoding, you can see the advertise-address in the cert.

**Reply by YuQiangZ** [reply]
1.刚刚按照您给的方式查看了，确实查到了，万分感谢！！！

2.在查看kubeedge命名空间的的secret时，发现有三个证书相关的（casecret    cloudcore    cloudcoresecret），并且发现cloudcore被挂载到了cloudcore容器内：
`        volumeMounts:
        - mountPath: /etc/kubeedge/config
          name: conf
        - mountPath: /etc/kubeedge
          name: certs
        - mountPath: /var/lib/kubeedge
          name: sock
        - mountPath: /etc/localtime
          name: host-time
          readOnly: true
      volumes:
      - configMap:
          defaultMode: 420
          name: cloudcore
        name: conf
      - name: certs
        secret:
          defaultMode: 420
          items:
          - key: stream.crt
            path: certs/stream.crt
          - key: stream.key
            path: certs/stream.key
          - key: streamCA.crt
            path: ca/streamCA.crt
          secretName: cloudcore
`
请问该证书的作用是什么，为什么另外两个secret没有挂载到容器中，边端在请求云端证书的时候不是要使用到cloudcoresecret中的证书吗？ @vincentgoat

**Reply by vincentgoat** [reply]
The edge cert will be generated automatically when connecting. The casecret host the ca cert.

**Comment by vincentgoat** [comment, marked as answer]
FYI: https://www.bilibili.com/video/BV1ze4y1u7bP/?spm_id_from=333.337.search-card.all.click

**Reply by YuQiangZ** [reply]
thank you!!! i'm trying
